### PR TITLE
DataFrame: move labels to field

### PR DESCRIPTION
**Pull request by ryantxu**
See #19755 

Move DataFrame labels from the root to each field

### CODE REVIEW

**Comment by ryantxu** [comment]
@aocenas -- can you take a look at this?  It seems to work OK in my testing, but I want to make sure it gets solid testing from people using tag based queries more often then me :)

**Comment by aocenas** [comment]
@ryantxu added questions in the issues.

**Comment by ryantxu** [comment]
Discussed with @aocenas -- we agree this is a good step forward, and just need to verify that everything works as expected for loki/prometheus

File: public/app/features/dashboard/state/DashboardMigrator.ts [hunk 1]
```diff
@@ -463,6 +463,28 @@ export class DashboardMigrator {
       });
     }
 
+    if (oldVersion < 21) {
+      const updateLinks = (link: DataLink) => {
+        return {
+          ...link,
+          url: link.url.replace(/__series.labels/g, '__field.labels'),
```
Comment (aocenas) [inline public/app/features/dashboard/state/DashboardMigrator.ts]: I think this should be safer with `/${__series.labels}/` so we do not replace something that is not a variable. Or is there multiple syntaxes for this? I know that there are for dashboard variables but thought in datalink only `${}` is allowed.

Comment (ryantxu) [reply public/app/features/dashboard/state/DashboardMigrator.ts]: @dprokop  ?

**Comment by marefr** [comment]
Would be good to get this merged today since it has quite a lot of changes and if you want it included in Grafana v6.5?

**Comment by aocenas** [comment]
@marefr 
> Would be good to get this merged today since it has quite a lot of changes and if you want it included in Grafana v6.5?

Looks fine though not sure if this needs to go to 6.5 thought it is ok even if this does not make the release. @ryantxu

**Comment by ryantxu** [comment]
> not sure if this needs to go to 6.5

please 6.5 :)  

It is the last *major* structural change to DataFrame I see for a while , and would love to have it included with the other semi-breaking changes in this release

**Review by aocenas — APPROVED** [review]

**Comment by aocenas** [comment]
@ryantxu ok went through the code and tested it a bit I assume right not this has a chance to affect only explore/loki parts right?

**Pull request merged**